internal/core/application/operator_service.go

```diff
@@ -814,6 +827,293 @@ func (o *operatorService) FeeAccountBalance(ctx context.Context) (
 	return int64(baseAssetAmount), nil
 }
 
+type depositType string
+
+var (
+	marketDeposit depositType = "MarketDeposit"
+	feeDeposit    depositType = "FeeDeposit"
+)
+
+func (o *operatorService) ClaimMarketDeposit(
+	ctx context.Context,
+	market Market,
+	outpoints []TxOutpoint,
+) error {
+	addressesPerAccount := make(map[int][]string)
+	bkPairs := make([][]byte, 0)
+	var accountIndex int
+	var err error
+
+	existingMarketReq := existingMarketReq(market)
+	if existingMarketReq {
+		if err := validateAssetString(market.BaseAsset); err != nil {
+			return domain.ErrInvalidBaseAsset
+		}
+
+		if err := validateAssetString(market.QuoteAsset); err != nil {
+			return domain.ErrInvalidQuoteAsset
+		}
+
+		if market.BaseAsset != config.GetString(config.BaseAssetKey) {
+			return domain.ErrInvalidBaseAsset
+		}
+
+		_, accountIndex, err = o.marketRepository.GetMarketByAsset(
+			ctx,
+			market.QuoteAsset,
+		)
+		if err != nil {
+			return err
+		}
+		if accountIndex != -1 {
+			addresses, bk, err := o.vaultRepository.
+				GetAllDerivedAddressesAndBlindingKeysForAccount(
+					ctx,
+					accountIndex,
+				)
+			if err != nil {
+				return err
+			}
+			addressesPerAccount[accountIndex] = addresses
+			bkPairs = bk
+		} else {
+			return domain.ErrMarketNotExist
+		}
+	} else {
```

Comment:
What  happens if the given market's asset hashes are empty? I do think we shoudl just return error to user. He will need to call the DepsitMarket first to create a new entry and fund it.

Reply:
if asset hashes are emty this means that Market is still not funded but DepositMarket is called